lib/text/ui_text_displayer.js

```diff
@@ -320,9 +325,19 @@ shaka.text.UITextDisplayer = class {
       style = span.style;
     }
 
-    style.backgroundColor = cue.backgroundColor;
-    style.border = cue.border;
+    if (isLeaf) {
+      style.backgroundColor = cue.backgroundColor;
+    }
+    if (isLeaf && parentCue && cue.backgroundColor == '') {
+      style.backgroundColor = parentCue.backgroundColor;
```

Comment:
This really seems like something that should be solved with style inheritance instead.  Do you have any idea why the element doesn't already inherit the parent's background and color styles?

Reply:
The problem is that the styles defined here (https://github.com/google/shaka-player/blob/master/ui/less/containers.less#L229-L235) overwrite those of the parent element (background-color and color)